internal/args/unmarshal.go

```diff
@@ -253,15 +220,50 @@ func set(dest reflect.Value, argNameWords []string, value string) error {
 			return &MissingStructFieldError{Dest: dest.Interface()}
 		}
 
-		// try to find the correct field in the struct.
+		// We cannot rely on dest.GetFieldByName() as reflect library is doing deep traversing when using anonymous field.
+		// Because of that we should rely on our own logic
+		//
+		// - First we try to find a field with the correct name in the current struct
+		// - If it does not exist we try to find it in all nested anonymous field
+		//   Anonymous fields are traversed from last to first as the last one in the struct declaration should take precedence
+
+		// We construct two cache:
+		anonymousFieldIndex := []int(nil)
+		fieldIndexByName := map[string]int{}
+		for i := 0; i < dest.Type().NumField(); i++ {
+			field := dest.Type().Field(i)
+			if field.Anonymous {
+				anonymousFieldIndex = append(anonymousFieldIndex, i)
+			} else {
+				fieldIndexByName[field.Name] = i
+			}
+		}
+
+		// Try to find the correct field in the current struct.
 		fieldName := strcase.ToPublicGoName(argNameWords[0])
-		field := dest.FieldByName(fieldName)
-		if !field.IsValid() {
-			return &UnknownArgError{}
+		if fieldIndex, exist := fieldIndexByName[fieldName]; exist {
+			return set(dest.Field(fieldIndex), argNameWords[1:], value)
+		}
+
+		//  If it do not exist we try to find it in nested anonymous field
+		for i := len(anonymousFieldIndex) - 1; i >= 0; i-- {
+			err := set(dest.Field(anonymousFieldIndex[i]), argNameWords, value)
```

Comment:
can't we first check the type of the field, instead of blindly trying all the anonymous fields ?

Reply:
no, because this needs to be recursive as an anonymousField may alson have anonymousField and so on,.